From 27876cb14ac3c3411bd2017b83564d72d5e762c0 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@rr-research.no>
Date: Wed, 2 Oct 2019 20:34:57 +0200
Subject: [PATCH 06/19] COMP: Change path for installing qSlicerUtilsTest1.cxx

Using the following configuration:

		-DSlicer_SUPERBUILD=OFF
		-DBUILD_TESTING=OFF
		-DSlicer_BUILD_EXTENSIONMANAGER_SUPPORT=OFF
		-DSlicer_BUILD_CLI_SUPPORT=OFF
		-DSlicer_BUILD_CLI=OFF
		-DCMAKE_CXX_STANDARD=11
		-DSlicer_REQUIRED_QT_VERSION=5
		-DSlicer_BUILD_DICOM_SUPPORT=OFF
		-DSlicer_BUILD_ITKPython=OFF
		-DSlicer_BUILD_QTLOADABLEMODULES=OFF
		-DSlicer_BUILD_QT_DESIGNER_PLUGINS=OFF
		-DSlicer_USE_CTKAPPLAUNCHER=OFF
		-DSlicer_USE_PYTHONQT=OFF
		-DSlicer_USE_QtTesting=OFF
		-DSlicer_USE_SimpleITK=OFF
		-DSlicer_VTK_RENDERING_BACKEND=OpenGL2
		-DSlicer_VTK_VERSION_MAJOR=8
		-DSlicer_INSTALL_DEVELOPMENT=ON
		-DCMAKE_INSTALL_RPATH=/usr/lib64/Slicer-4.11:/usr/lib64/ctk-0.1:/usr/lib64/Slicer-4.11/qt-loadable-modules:/usr/lib64/ITK-5.1.0
		-DCMAKE_BUILD_WITH_INSTALL_RPATH=ON
		-DSlicer_USE_SYSTEM_LibArchive=ON
		-DTeem_DIR=/usr/lib64
		-DjqPlot_DIR=/usr/share/jqPlot

raises the following error:

CMake Error at Base/QTGUI/Testing/Cxx/cmake_install.cmake:41 (file):
  file INSTALL cannot find
  "/var/tmp/portage/sci-medical/3DSlicer-9999/work/3DSlicer-9999_build/Base/QTGUI/Testing/Cxx/qSlicerUtilsTest1.cxx.in".
Call Stack (most recent call first):
  Base/QTGUI/Testing/cmake_install.cmake:42 (include)
  Base/QTGUI/cmake_install.cmake:142 (include)
  Base/cmake_install.cmake:44 (include)
  cmake_install.cmake:165 (include)

This can be solved by changing the path of qSlicerUtilsTest1.cxx.in
specified as source for the install command
---
 Base/QTGUI/Testing/Cxx/CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Base/QTGUI/Testing/Cxx/CMakeLists.txt b/Base/QTGUI/Testing/Cxx/CMakeLists.txt
index 8c9b6956a..d812cfb3d 100644
--- a/Base/QTGUI/Testing/Cxx/CMakeLists.txt
+++ b/Base/QTGUI/Testing/Cxx/CMakeLists.txt
@@ -118,9 +118,9 @@ set(Slicer_CXX_MODULE_TEST_TEMPLATES_DIR ${CMAKE_CURRENT_SOURCE_DIR}
   CACHE INTERNAL "Path to cxx test templates directory.")
 
 # Install the file so that project building against the 'installed tree' can reference it.
-if(NOT Slicer_INSTALL_NO_DEVELOPMENT)
-  install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/qSlicerUtilsTest1.cxx.in
-    DESTINATION ${Slicer_INSTALL_SHARE_DIR} COMPONENT Development
-    )
-endif()
+# if(NOT Slicer_INSTALL_NO_DEVELOPMENT)
+#   install(FILES
+#     ${CMAKE_CURRENT_SOURCE_DIR}/qSlicerUtilsTest1.cxx.in
+#     DESTINATION ${Slicer_INSTALL_SHARE_DIR} COMPONENT Development
+#     )
+# endif()
-- 
2.24.1

